src/event.rs

```diff
@@ -353,8 +353,13 @@ pub fn process_event(ctx: &mut Context, event: &winit::event::Event<()>) {
                 };
                 ctx.keyboard_context.set_key(*keycode, pressed);
             }
-            winit_event::WindowEvent::ScaleFactorChanged { .. } => {
-                // Nope.
+            winit_event::WindowEvent::ScaleFactorChanged { new_inner_size, .. } => {
+                // actively set the new_inner_size to be the desired size
+                // to stop winit from resizing our window
+                **new_inner_size = winit::dpi::PhysicalSize::<u32>::from([
+                    ctx.conf.window_mode.width,
+                    ctx.conf.window_mode.height,
+                ]);
```

Comment:
> Do we really want to implement a workaround for this? Or do we just want to label it a dependency bug and move on?

I, for one, think we should add a configuration option like `resize_on_screen_scale_factor_changed` to some sub-context which is checked before executing this.

If set to true, then this fix doesn't get executed and ggez will show the old behavior (i.e. ggez will give you a more "pleasing" screen change behavior, which might break your game if you don't plan for it).
Therefore this value should by default be set to false.